src/util/__tests__/importRoam.ts

```diff
@@ -0,0 +1,246 @@
+              'create-email': 'testemail@gmail.com',
+              'create-time': 1600111383054,
+              children: [
+                {
+                  string: 'C',
+                  'create-email': 'testemail@gmail.com',
+                  'create-time': 1600111383911,
+                  uid: 'HMN_YQtZZ',
+                }],
+              uid: 'JBXKlMcxh',
+            }],
+          uid: '0VQBPmUiy',
+        }],
+    },
+    {
+      title: 'September 5th, 2020',
+      children: [
+        {
+          string: 'X',
+          'create-email': 'testemail@gmail.com',
+          'create-time': 1600111456859,
+          children: [
+            {
+              string: 'Y',
+              'create-email': 'testemail@gmail.com',
+              'create-time': 1600111457621,
+              children: [
+                {
+                  string: '[[September 4th, 2020]]',
+                  'create-email': 'testemail@gmail.com',
+                  'create-time': 1600111458385,
+                  uid: 'Wt5NR3b56',
+                }],
+              uid: 'obXRTMWqJ',
+            }],
+          uid: 'Pu444IoIi',
+        }],
+    }]
+  const res = importExport(testData)
+  expect(res)
+    .toBe(`
+- September 4th, 2020
+  - A
+    - B
+      - C
+- September 5th, 2020
+  - X
+    - Y
+      - [[September 4th, 2020]]
+`)
+})
+
+test('it should save create-time as created property', () => {
+  const testData = [
+    {
+      title: 'Fruits',
+      children: [
+        {
+          string: 'Apple',
+          'create-email': 'testemail@gmail.com',
+          'create-time': 1600111381583,
+          'edit-time': 1600111381580,
+          uid: 'UK11200',
+        },
+        {
+          string: 'Orange',
+          'create-email': 'testemail@gmail.com',
+          'create-time': 1600111383054,
+          'edit-time': 1600111383050,
+          uid: 'UK11233',
+        },
+        {
+          string: 'Banana',
+          'create-email': 'testemail@gmail.com',
+          'create-time': 1600111383911,
+          'edit-time': 1600111383910,
+          uid: 'HMN_YQtZZ',
+        }
+      ],
+    },
+    {
+      title: 'Veggies',
+      children: [
+        {
+          string: 'Broccoli',
+          'create-email': 'testemail@gmail.com',
+          'create-time': 1600111381600,
+          'edit-time': 1600111381599,
+          uid: 'BK11200',
+        },
+        {
+          string: 'Spinach',
+          'create-email': 'testemail@gmail.com',
+          'create-time': 1600111389054,
+          'edit-time': 1600111389050,
+          uid: 'BK11233',
+        }
+      ],
+    }
+  ]
+
+  const roamBlocks = [...testData[0].children, ...testData[1].children]
+
+  const {
+    thoughtIndexUpdates: thoughtIndex,
+  } = importROAM(testState, RANKED_ROOT as SimplePath, testData)
+
+  Object.keys(thoughtIndex)
+    // ignore root blocks of ROAM page since they won't have create/edit time
+    .filter((_, index) => index !== 0 && index !== 4)
```

Comment:
Matching by index here is quite fragile. If the test changes at all it will break, or worse create a false positive. In general we want to avoid creating dependencies between things that are not semantically related, or only incidentally related. 

Instead, I would suggest filtering out based on the key.